### PR TITLE
[IMP] base: currency changes according to the ISO 4217 standard

### LINKED ISSUE #63863: Change Default value on Currency in KRW
Impacted versions: All

Steps to reproduce:

Current behavior: 

-  Decimal = 2
-  Position = after

Expected behavior: 

- Decimal = 0
- Potion = Before

Video/Screenshot link (optional):
image

Support ticket number submitted via odoo.com/help (optional):

### PR DESCRIPTION
**Pull request by ppr-odoo**
ISO 4217 standard according to:
                 -   Added new currency
                 -  Update the name and rounding of the currency
 Removed a currency that has been obsolete for 10 or more years. 

Task-Id: 2501384

### CODE REVIEW

File: odoo/addons/base/data/res_currency_data.xml [hunk 1]
```diff
@@ -140,7 +140,7 @@
 
         <record id="XPF" model="res.currency">
             <field name="name">XPF</field>
-            <field name="full_name">CFP franc</field>
+            <field name="full_name">CFP franc (franc Pacifique)</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:143]:
```suggestion
            <field name="full_name">CFP franc</field>
```
I don't believe that the parentheses are part of the name.

File: odoo/addons/base/data/res_currency_data.xml [hunk 2]
```diff
@@ -170,7 +170,7 @@
 
         <record id="SEK" model="res.currency">
             <field name="name">SEK</field>
-            <field name="full_name">Swedish krona</field>
+            <field name="full_name">Swedish krona/kronor</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:173]:
```suggestion
            <field name="full_name">Swedish krona</field>
```

File: odoo/addons/base/data/res_currency_data.xml [hunk 3]
```diff
@@ -839,21 +819,11 @@
             <field name="currency_subunit_label">Penny</field>
         </record>
 
-        <record id="QTQ" model="res.currency">
-            <field name="name">QTQ</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:843]: This one still exists but its code is GTQ.

File: odoo/addons/base/data/res_currency_data.xml [hunk 4]
```diff
@@ -1327,16 +1287,6 @@
             <field name="currency_subunit_label">Fils</field>
         </record>
 
-        <record id="UAG" model="res.currency">
-            <field name="name">UAG</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1331]: This one still exists but its code is UAH.

File: odoo/addons/base/data/res_currency_data.xml [hunk 5]
```diff
@@ -1687,34 +1607,146 @@
             <field name="currency_subunit_label">Centavo</field>
         </record>
 
-        <record id="SKK" model="res.currency">
-            <field name="name">SKK</field>
-            <field name="full_name">Slovak koruna</field>
-            <field name="symbol">Sk</field>
+        <record id="GTQ" model="res.currency">
+            <field name="name">GTQ</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1611]: See my comment above.

File: odoo/addons/base/data/res_currency_data.xml [hunk 6]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="currency_unit_label">Quetzal</field>
+            <field name="currency_subunit_label">Centavo</field>
+        </record>
+
+        <record id="XDR" model="res.currency">
+            <field name="name">XDR</field>
+            <field name="full_name">Special drawing rights</field>
+            <field name="symbol">SDR</field>
+            <field name="active" eval="False"/>
+        </record>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1625]: I don't think that we should add special currencies.

**Comment by pedrobaeza** [comment]
What about adding also the numeric code (and maybe full name) defined in ISO 4217 (https://en.wikipedia.org/wiki/ISO_4217) to the res.currency model for avoiding extra modules like https://github.com/OCA/community-data-files/tree/14.0/base_currency_iso_4217 ?

File: odoo/addons/base/data/res_currency_data.xml [hunk 7]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="rounding">0.0001</field>
+            <field name="active" eval="False"/>
+            <field name="position">before</field>
+            <field name="currency_unit_label">Peso</field>
+            <field name="currency_subunit_label">Centavos</field>
+        </record>
+
+        <record id="BOV" model="res.currency">
+            <field name="name">BOV</field>
+            <field name="full_name">Bolivian Mvdol (funds code)</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1651]:
```suggestion
            <field name="full_name">Bolivian Mvdol</field>
```

File: odoo/addons/base/data/res_currency_data.xml [hunk 8]
```diff
@@ -1687,34 +1607,146 @@
...
+
+        <record id="USN" model="res.currency">
+            <field name="name">USN</field>
+            <field name="full_name">United States dollar (next day)</field>
+            <field name="symbol">$</field>
+            <field name="rounding">0.1</field>
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Dollars</field>
+            <field name="currency_subunit_label">Cents</field>
+        </record>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1704]: As I understand it, it's a particular case of the USD. Probably not worth adding.

File: odoo/addons/base/data/res_currency_data.xml (hunk 7 above)
Comment (AntoineVDV) [reply odoo/addons/base/data/res_currency_data.xml:1651]: In fact, such special currencies shouldn't be added. See my comment on USN below.

File: odoo/addons/base/data/res_currency_data.xml [hunk 9]
```diff
@@ -1687,34 +1607,146 @@
...
+
+        <record id="STN" model="res.currency">
+            <field name="name">STN</field>
+            <field name="full_name">São Tomé and Príncipe dobra</field>
+            <field name="symbol">Db</field>
+            <field name="rounding">0.1</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1710]:
```suggestion
            <field name="rounding">0.01</field>
```

File: odoo/addons/base/data/res_currency_data.xml [hunk 10]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Dobra</field>
+            <field name="currency_subunit_label">cêntimo</field>
+        </record>
+
+        <record id="MXV" model="res.currency">
+            <field name="name">MXV</field>
+            <field name="full_name">Mexican Unidad de Inversion</field>
+            <field name="symbol">$</field>
+            <field name="rounding">0.1</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1720]:
```suggestion
            <field name="rounding">0.01</field>
```

Comment (AntoineVDV) [reply odoo/addons/base/data/res_currency_data.xml:1720]: Again, not worth adding as it is a funds code.

File: odoo/addons/base/data/res_currency_data.xml [hunk 11]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Pesos</field>
+            <field name="currency_subunit_label">Centavos</field>
+        </record>
+
+        <record id="CHW" model="res.currency">
+            <field name="name">CHW</field>
+            <field name="full_name">WIR franc (complementary currency)</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1728]:
```suggestion
            <field name="full_name">WIR franc</field>
```

Comment (AntoineVDV) [reply odoo/addons/base/data/res_currency_data.xml:1728]: Not sure about this one.. Let's see what @mart-e has to say about it.

File: odoo/addons/base/data/res_currency_data.xml [hunk 12]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="symbol">fr</field>
+            <field name="rounding">0.1</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1730]:
```suggestion
            <field name="rounding">0.01</field>
```

File: odoo/addons/base/data/res_currency_data.xml [hunk 13]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Franc</field>
+            <field name="currency_subunit_label">Centimes</field>
+        </record>
+
+        <record id="CHE" model="res.currency">
+            <field name="name">CHE</field>
+            <field name="full_name">WIR euro (complementary currency)</field>
+            <field name="symbol">fr</field>
+            <field name="rounding">0.1</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1740]:
```suggestion
            <field name="rounding">0.01</field>
```

File: odoo/addons/base/data/res_currency_data.xml [hunk 14]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Franc</field>
+            <field name="currency_subunit_label">Centimes</field>
+        </record>
+
+        <record id="CHE" model="res.currency">
+            <field name="name">CHE</field>
+            <field name="full_name">WIR euro (complementary currency)</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1738]:
```suggestion
            <field name="full_name">WIR euro</field>
```
@mart-e Same as above.

File: odoo/addons/base/data/res_currency_data.xml [hunk 15]
```diff
@@ -1687,34 +1607,146 @@
...
+            <field name="rounding">0.1</field>
+            <field name="active" eval="False"/>
+            <field name="currency_unit_label">Rappen</field>
+        </record>
+
+        <record id="XUA" model="res.currency">
+            <field name="name">XUA</field>
+            <field name="full_name">ADB Unit of Account</field>
+            <field name="active" eval="False"/>
+            <field name="symbol">CFA</field>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:1749]: Probably not worth adding too. It's not a regular currency.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello, sorry for the late review. Please find my comments below :)

File: odoo/addons/base/data/res_currency_data.xml [hunk 16]
```diff
@@ -316,7 +316,7 @@
             <field name="name">KRW</field>
             <field name="full_name">South Korean won</field>
             <field name="symbol">₩</field>
-            <field name="rounding">0.01</field>
+            <field name="rounding">1</field>
             <field name="active" eval="False"/>
```
Comment (AntoineVDV) [inline odoo/addons/base/data/res_currency_data.xml:320]:
```suggestion
            <field name="active" eval="False"/>
            <field name="position">before</field>
```
To address issue https://github.com/odoo/odoo/issues/63863

**Comment by AntoineVDV** [comment]
> What about adding also the numeric code (and maybe full name) defined in ISO 4217 (https://en.wikipedia.org/wiki/ISO_4217) to the res.currency model for avoiding extra modules like https://github.com/OCA/community-data-files/tree/14.0/base_currency_iso_4217 ?

Hello, this seems like a good idea but this PR is only about fixing inconsistencies. I suggest opening a new issue to discuss the addition of two new fields in the `res.currency` table and ping the relevant people. `@mart-e` and `@jco-odoo` I suppose.

**Comment by ppr-odoo** [comment]
@AntoineVDV Changes done :+1: 
Thanks

**Review by AntoineVDV — APPROVED** [review]
Thanks for your work :+1: (still waiting for mat's review)

Please update the commit message as follows:

> [IMP] base, l10n_(ar, cl): update currency data with ISO 4217 standard
> 
> This commit updates the `res.currency` data files to be compliant with
the ISO 4217 standard:
> - Add existing currencies missing from Odoo.
> - Remove currencies deprecated since at least 10 years.
> - Update incorrect currency names and rounding values.
> 
> task-2501384

**Review by mart-e — APPROVED** [review]

File: odoo/addons/base/data/res_currency_data.xml (hunk 11 above)
Comment (mart-e) [reply odoo/addons/base/data/res_currency_data.xml]: As much as I love the ideology of local monetary systems, I don't think this is useful to add them in the base list and is the open door to all the city-based currencies or worse... cryptocurrencies :scream: 
I could see that in l10n_ch if somebody is asking.

File: odoo/addons/base/data/res_currency_data.xml (hunk 14 above)
Comment (mart-e) [reply odoo/addons/base/data/res_currency_data.xml]: Same, I could accept this is a localisation module if explicitly asked but I don't think we need this in base.

Comment (ppr-odoo) [reply odoo/addons/base/data/res_currency_data.xml]: @mart-e @AntoineVDV @mba-odoo Changes done

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**